mesonbuild/modules/gnome.py

```diff
@@ -776,7 +776,28 @@ def gtkdoc(self, state, args, kwargs):
         args += self._unpack_args('--fixxrefargs=', 'fixxref_args', kwargs)
         args += self._unpack_args('--mkdbargs=', 'mkdb_args', kwargs)
         args += self._unpack_args('--html-assets=', 'html_assets', kwargs, state)
-        args += self._unpack_args('--content-files=', 'content_files', kwargs, state)
+
+        content_files = []
+        for s in mesonlib.extract_as_list(kwargs, 'content_files'):
+            if hasattr(s, 'held_object'):
+                s = s.held_object
+            if isinstance(s, (build.CustomTarget, build.CustomTargetIndex)):
+                content_files.append(os.path.join(state.environment.get_build_dir(),
+                                                  state.backend.get_target_dir(s),
+                                                  s.get_outputs()[0]))
+            elif isinstance(s, mesonlib.File):
+                content_files.append(s.rel_to_builddir(state.build_to_src))
+            elif isinstance(s, build.GeneratedList):
+                for gen_src in s.get_outputs():
+                    content_files.append(os.path.join(state.environment.get_source_dir(),
+                                                      state.subdir,
+                                                      gen_src))
+            else:
```

Comment:
This should be `elif isinstance(s, str)` and the `else:` condition should be a `MesonException` that an invalid object was passed, with a good error message.